src/mergeTypeDefs.ts

```diff
@@ -7,42 +7,50 @@ import { ITypeDefinitions } from 'graphql-tools';
  * @param type Operation type to merge
  */
 function merge(defs: ITypeDefinitions, type: String): ITypeDefinitions {
-  let operations = [];
+  let result;
 
   defs = (defs as any).filter(def => {
-    const definitions = def.definitions.filter(definition => {
-      const name = definition.name;
-      if (definition.kind === 'ObjectTypeDefinition' && name.kind === 'Name' && name.value === type) {
-        operations.push(definition);
+    // filter out falsy values just incase
+    if (!def) return false;
+
+    // reset def.definitions to be the filtered version so the `type` that
+    // was passed in doesn't exist in it anymore
+    def.definitions = def.definitions.filter(definition => {
+      const { kind, name, fields } = definition;
+      if (kind === 'ObjectTypeDefinition' && name.kind === 'Name' && name.value === type) {
+        if (!result) {
+          result = definition;
+        } else {
+          result.fields = result.fields.concat(fields);
+        }
+
+        // remove the definition because it's been added to the result and
+        // shouldn't exist here anymore
         return false;
       }
 
       return true;
     });
 
-    if (definitions.length === 0) {
+    if (def.definitions.length === 0) {
       return false; // operation was the only definition so this can be removed
     }
 
     return true;
   });
 
-  if (operations.length > 0) {
-    if (operations.length > 1) {
-      operations = operations.reduce((acc, query) => {
-        if (acc === undefined) {
-          acc = query;
-        } else {
-          acc.fields = acc.fields.concat(query.fields);
-        }
-
-        return acc;
+  // update to be `result?.fields?.length` when it's available
+  if (((result || {}).fields || []).length) {
```

Comment:
this is doing the same thing it was doing before, it's just like this because I merged the array of fields above `result.fields = result.fields.concat(fields);`. The only reason why I made it `result?.fields?.length` is because the type that the user passed in might not exist so the `result` would still be `undefined`